cpp/src/arrow/util/bit_util_test.cc

```diff
@@ -1589,16 +1589,19 @@ TEST(BitUtilTests, TestCopyBitmap) {
 
   std::vector<int64_t> lengths = {kBufferSize * 8 - 4, kBufferSize * 8};
   std::vector<int64_t> offsets = {0, 12, 16, 32, 37, 63, 64, 128};
-  for (int64_t num_bits : lengths) {
-    for (int64_t offset : offsets) {
-      const int64_t copy_length = num_bits - offset;
+  for (bool maintan_offset : {true, false}) {
+    for (int64_t num_bits : lengths) {
+      for (int64_t offset : offsets) {
+        const int64_t copy_length = num_bits - offset;
```

Comment:
It would be better to test with different offsets, for example:
```suggestion
  for (int64_t num_bits : lengths) {
    for (int64_t offset : offsets) {
      for (int64_t dest_offset : {0, offset - 1, offset + 1}) {
        const int64_t copy_length = num_bits - offset;
```